src/java/org/apache/cassandra/index/sai/disk/io/IndexInputReader.java

```diff
@@ -72,36 +75,6 @@ public void readBytes(byte[] bytes, int off, int len) throws IOException
         input.readFully(bytes, off, len);
     }
 
-    /**
-     * Using {@link RandomAccessReader#readShort()} directly is faster than {@link DataInput#readShort()} which calls
-     * {@link DataInput#readByte()} one by one
-     */
-    @Override
-    public short readShort() throws IOException
-    {
-        return input.readShort();
-    }
-
-    /**
-     * Using {@link RandomAccessReader#readInt()} directly is faster than {@link DataInput#readInt()} which
-     * calls {@link DataInput#readByte()} one by one
-     */
-    @Override
-    public int readInt() throws IOException
-    {
-        return input.readInt();
-    }
-
-    /**
-     * Using {@link RandomAccessReader#readLong()} directly is faster than {@link DataInput#readLong()} which
-     * calls {@link DataInput#readByte()} one by one
-     */
-    @Override
-    public long readLong() throws IOException
-    {
-        return input.readLong();
-    }
-
```

Comment:
Are these optimizations no longer necessary? it seems like `DataInput` still reads byte-by-byte...

Do we just never use these methods?

CC @mike-tr-adamson

Reply:
those are necessary because of the comment that you wanted to turn into a javadoc

simpler and probably more performant to remove them and let the parent class do its thing, than to override and add in a reverseBytes